### PR TITLE
rename/deprecate block_size -> block_copy_size, improve docs

### PR DESCRIPTION
**Pull request by Wumpf**
**Checklist**

- [x] Run `cargo clippy`.
- [ ] Run `cargo clippy --target wasm32-unknown-unknown` if applicable.
- [x] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
* Very loosely related to #4054

**Description**
I keep seeing people getting confused about block_size, so I made its name more explicit and fixed the docs a bit (for instance the link in there was broken)

### CODE REVIEW

**Comment by teoxoy** [comment]
`texel_block_size` has been renamed to `texel_block_copy_footprint` in the spec which is a lot clearer in what it means. Can we use that instead?

https://github.com/gpuweb/gpuweb/pull/3922

**Comment by Wumpf** [comment]
Oh didn't notice, thank you! Let's go with that then and link to it ofc.
That's a pretty awkward name though imho. Also doesn't imply bytes -.-

**Comment by Wumpf** [comment]
actually.. linking it okay. But do we think anyone will ever find `TextureFormat::block_copy_footprint` if they're looking for a byte size of a texel?

**Comment by Wumpf** [comment]
Technically we are providing the informative value that is described in the spec as

> The texel block memory cost of a [GPUTextureFormat](https://gpuweb.github.io/gpuweb/#enumdef-gputextureformat) is the number of bytes needed to store one [texel block](https://gpuweb.github.io/gpuweb/#texel-block). It is not fully defined for all formats. This value is informative and non-normative.


so from that pov we would need to call it block_memory_cost :/

**Comment by Wumpf** [comment]
moving (potential) discussion to Matrix

**Comment by teoxoy** [comment]
The texel block memory cost is only "informative and non-normative".
The texel block copy footprint is the useful value since it's needed to calculate some of the parameters of `ImageDataLayout`/`ImageCopyBuffer`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ For naga changelogs at or before v0.14.0. See [naga's changelog](naga/CHANGELOG.
 #### General
 
 - Log vulkan validation layer messages during instance creation and destruction: By @exrook in [#4586](https://github.com/gfx-rs/wgpu/pull/4586)
+- `TextureFormat::block_size` is deprecated, use `TextureFormat::block_copy_footprint` instead: By @wumpf in [#4647](https://github.com/gfx-rs/wgpu/pull/4647)
```
Comment (teoxoy) [inline CHANGELOG.md]:
```suggestion
- `TextureFormat::block_size` is deprecated, use `TextureFormat::block_copy_size` instead: By @wumpf in [#4647](https://github.com/gfx-rs/wgpu/pull/4647)
```

**Review by teoxoy — APPROVED** [review]
LGTM

**Pull request merged**